### PR TITLE
16-BUNSPACE2024-Create-Footer-And-NavButtonFor Q&A

### LINKED ISSUE #16: Create footer with a basic logo and a navbutton for a q&a

### PR DESCRIPTION
**Pull request by GrpFruit**
Add footer and contact button

Fix footer layout

## Extra Resources

- [The 10 commandments of navigating code reviews](https://techbeacon.com/app-dev-testing/10-commandments-navigating-code-reviews)
- [Google's Code Review Developer Guide](https://google.github.io/eng-practices/review/)

Closes #16

### CODE REVIEW

File: ui/src/Footer/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+];
+export function Footer() {
+  return (
+    <footer className="footer">
+      <a className="footer__img" href="/home">
+        <img src="peachy.png" width="55" height="55"/>
+      </a>
+      {footerButtons.map((button) => {
+        return (
+          <a className="footer__button" href={button.path} key={button.label}>
```
Comment (jmorales8) [inline ui/src/Footer/Footer.tsx:17]: NavBar looks good, but have the buttons be a different style than the navbar buttons. Up to you how you want to style them. But have them be unique

File: ui/src/App.scss [hunk 2]
```diff
@@ -5,3 +5,11 @@
   overflow: hidden;
   position: fixed;
 }
+.footer {
```
Comment (jmorales8) [inline ui/src/App.scss:8]: Why do we declare two classes of .footer. We declare one here and the other in footer.scss

**Comment by jmorales8** [comment]
Rebase this branch with master, i want to see what this looks like and behaves when combines with the newly added scrollbar

File: ui/src/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+interface FooterButtons {
+  label: string;
+  path: string;
+}
+
+const footerButtons: FooterButtons[] = [
+  { label: "Q&A", path: "/QA" },
```
Comment (jmorales8) [inline ui/src/Footer/Footer.tsx:7]: Talked to pillow, The path should be `/Q-and-A`

Comment (GrpFruit) [reply ui/src/Footer/Footer.tsx:7]: Talked to pillow is the label? I can fix the path!

File: ui/src/App.scss (hunk 2 above)
Comment (GrpFruit) [reply ui/src/App.scss]: Fixed and removed

**Comment by GrpFruit** [comment]
> Rebase this branch with master, i want to see what this looks like and behaves when combines with the newly added scrollbar

The footer not working correctly trying to fix

File: ui/src/Footer/Footer.tsx (hunk 3 above)
Comment (jmorales8) [reply ui/src/Footer/Footer.tsx:7]: Talked to pillow in terms of Pillow is her name, the vtuber character. Just put the path as the new one and leave the label alone

**Comment by GrpFruit** [comment]
> > Rebase this branch with master, i want to see what this looks like and behaves when combines with the newly added scrollbar
> 
> The footer not working currently trying to fix

@jmorales8 is working

**Comment by jmorales8** [comment]
Looks good +1

File: ui/src/Footer/footer.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    &:hover{
+      background-color: rgb(255, 243, 245);
+      border: 1px solid hotpink;
+    }
+  }
+}
+
+#footer {
+  
+}
```
Comment (jmorales8) [inline ui/src/Footer/footer.scss:32]: Get rid of this and add a new line and its good!

**Comment by jmorales8** [comment]
Looks good, all its missing is the active with the footer button so and its good to merge

File: ui/src/App.scss [hunk 5]
```diff
@@ -1,16 +1,27 @@
...
   background-color: #eecbcb;
   overflow-y: auto;
   overflow-x: hidden;
-  position: fixed;
+  &__content {
+    margin-bottom: 5rem;
+  }
+}
+
+body, #root {
```
Comment (jmorales8) [inline ui/src/App.scss:13]: Why was this added, i removed it and noticed 0 difference

File: ui/src/App.scss [hunk 6]
```diff
@@ -1,16 +1,27 @@
 .app {
-  width: 100%;
-  height: 100%;
+  display: flex;
+  flex-direction: column;
+  min-height: 100vh;
   background-color: #eecbcb;
   overflow-y: auto;
```
Comment (jmorales8) [inline ui/src/App.scss:6]: Get rid of this line, we no longer need it since `html {overflow-y: scroll}` does what we want, good job. I coudlnt even figure out this behavior

File: ui/src/App.scss (hunk 5 above)
Comment (GrpFruit) [reply ui/src/App.scss:13]: We dont need it any more i have removed it

**Comment by jmorales8** [comment]
Looks good +1
